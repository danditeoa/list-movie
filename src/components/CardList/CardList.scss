.list-container {
  display: flex;
  flex-wrap: wrap;
  width: 1000px;
  align-items: center;
  justify-content: center;
}