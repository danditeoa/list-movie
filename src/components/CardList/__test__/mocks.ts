export const mockList = [
  {
    adult: false,
    backdrop_path: '/y5Z0WesTjvn59jP6yo459eUsbli.jpg',
    genre_ids: [27, 53],
    id: 663712,
    original_language: 'en',
    original_title: 'Terrifier 2',
    overview:
      "After being resurrected by a sinister entity, Art the Clown returns to Miles County where he must hunt down and destroy a teenage girl and her younger brother on Halloween night.  As the body count rises, the siblings fight to stay alive while uncovering the true nature of Art's evil intent.",
    popularity: 7866.782,
    poster_path: '/b6IRp6Pl2Fsq37r9jFhGoLtaqHm.jpg',
    release_date: '2022-10-06',
    title: 'Terrifier 2',
    video: false,
    vote_average: 7.1,
    vote_count: 445,
  },
  {
    adult: false,
    backdrop_path: '/bQXAqRx2Fgc46uCVWgoPz5L5Dtr.jpg',
    genre_ids: [28, 14, 878],
    id: 436270,
    original_language: 'en',
    original_title: 'Black Adam',
    overview:
      'Nearly 5,000 years after he was bestowed with the almighty powers of the Egyptian gods—and imprisoned just as quickly—Black Adam is freed from his earthly tomb, ready to unleash his unique form of justice on the modern world.',
    popularity: 4428.609,
    poster_path: '/3zXceNTtyj5FLjwQXuPvLYK5YYL.jpg',
    release_date: '2022-10-19',
    title: 'Black Adam',
    video: false,
    vote_average: 6.9,
    vote_count: 905,
  },
  {
    adult: false,
    backdrop_path: '/pGx6O6IwqADOsgmqWzPysmWnOyr.jpg',
    genre_ids: [28, 14],
    id: 732459,
    original_language: 'en',
    original_title: 'Blade of the 47 Ronin',
    overview:
      'In this sequel to "47 Ronin," a new class of warriors emerges among the Samurai clans to keep a sought-after sword from falling into the wrong hands.',
    popularity: 3821.739,
    poster_path: '/kjFDIlUCJkcpFxYKtE6OsGcAfQQ.jpg',
    release_date: '2022-10-25',
    title: 'Blade of the 47 Ronin',
    video: false,
    vote_average: 6.7,
    vote_count: 47,
  },
  {
    adult: false,
    backdrop_path: '/naNXYdBzTEb1KwOdi1RbBkM9Zv1.jpg',
    genre_ids: [27, 53],
    id: 420634,
    original_language: 'en',
    original_title: 'Terrifier',
    overview:
      'On Halloween night, a young woman finds herself as the obsession of a sadistic murderer known as Art the Clown.',
    popularity: 3031.419,
    poster_path: '/6PQqC4SbY910VvyVad6mvsboILU.jpg',
    release_date: '2016-10-15',
    title: 'Terrifier',
    video: false,
    vote_average: 6.6,
    vote_count: 889,
  },
  {
    adult: false,
    backdrop_path: '/b2FxWOxe9K7ZZ1uaPOze2RJ1ajq.jpg',
    genre_ids: [27, 28, 35],
    id: 675054,
    original_language: 'es',
    original_title: 'MexZombies',
    overview:
      'A group of teenagers must face a zombie apocalypse, and help reestablish order.',
    popularity: 2784.235,
    poster_path: '/pTxw4GFE3ZfzJfvuUsPyD5njAuI.jpg',
    release_date: '2022-10-26',
    title: 'MexZombies',
    video: false,
    vote_average: 7.3,
    vote_count: 49,
  },
  {
    adult: false,
    backdrop_path: '/1DBDwevWS8OhiT3wqqlW7KGPd6m.jpg',
    genre_ids: [53],
    id: 985939,
    original_language: 'en',
    original_title: 'Fall',
    overview:
      'For best friends Becky and Hunter, life is all about conquering fears and pushing limits. But after they climb 2,000 feet to the top of a remote, abandoned radio tower, they find themselves stranded with no way down. Now Becky and Hunter’s expert climbing skills will be put to the ultimate test as they desperately fight to survive the elements, a lack of supplies, and vertigo-inducing heights',
    popularity: 2350.808,
    poster_path: '/spCAxD99U1A6jsiePFoqdEcY0dG.jpg',
    release_date: '2022-08-11',
    title: 'Fall',
    video: false,
    vote_average: 7.3,
    vote_count: 1650,
  },
];
