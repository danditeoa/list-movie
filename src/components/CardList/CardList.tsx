import React from 'react';
import { Movie } from '../../models/paginatedMovie.model';
import CardMovie from '../cardMovie/CardMovie';
import "./CardList.scss";

export type CardListProp= {
  list: Movie[]
} 
const CardList = ({list}: CardListProp) => {
  if(!list) {
  return null;
  }

  return (
      <div className="list-container">
        {list.map(movie =><CardMovie key={movie.id} movie={movie} />)}
      </div> 
  );
};

export default CardList;