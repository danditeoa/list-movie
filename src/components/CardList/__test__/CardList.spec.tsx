import { render, screen } from '@testing-library/react';
import CardList from '../CardList';
import { mockList } from './mocks';

describe('<CardList />', () => {
  it('matches snapshot', () => {
    const { container } = render(<CardList list={mockList} />);
    expect(container).toMatchSnapshot();
  });

  it('shows "Lista vazia!" when is empty list', () => {
    render(<CardList list={[]} />);
    expect(screen.getByText('Lista vazia!')).toBeInTheDocument();
  });
});
