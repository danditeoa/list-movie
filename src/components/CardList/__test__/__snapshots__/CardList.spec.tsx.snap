// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> matches snapshot 1`] = `
<div>
  <div
    class="list-container"
  >
    <div
      class="card-container"
    >
      <img
        alt="image backdrop"
        src="https://image.tmdb.org/t/p/w300//y5Z0WesTjvn59jP6yo459eUsbli.jpg"
      />
      <h2>
        Terrifier 2
      </h2>
      <p>
        <strong>
          2022
        </strong>
      </p>
    </div>
    <div
      class="card-container"
    >
      <img
        alt="image backdrop"
        src="https://image.tmdb.org/t/p/w300//bQXAqRx2Fgc46uCVWgoPz5L5Dtr.jpg"
      />
      <h2>
        Black Adam
      </h2>
      <p>
        <strong>
          2022
        </strong>
      </p>
    </div>
    <div
      class="card-container"
    >
      <img
        alt="image backdrop"
        src="https://image.tmdb.org/t/p/w300//pGx6O6IwqADOsgmqWzPysmWnOyr.jpg"
      />
      <h2>
        Blade of the 47 Ronin
      </h2>
      <p>
        <strong>
          2022
        </strong>
      </p>
    </div>
    <div
      class="card-container"
    >
      <img
        alt="image backdrop"
        src="https://image.tmdb.org/t/p/w300//naNXYdBzTEb1KwOdi1RbBkM9Zv1.jpg"
      />
      <h2>
        Terrifier
      </h2>
      <p>
        <strong>
          2016
        </strong>
      </p>
    </div>
    <div
      class="card-container"
    >
      <img
        alt="image backdrop"
        src="https://image.tmdb.org/t/p/w300//b2FxWOxe9K7ZZ1uaPOze2RJ1ajq.jpg"
      />
      <h2>
        MexZombies
      </h2>
      <p>
        <strong>
          2022
        </strong>
      </p>
    </div>
    <div
      class="card-container"
    >
      <img
        alt="image backdrop"
        src="https://image.tmdb.org/t/p/w300//1DBDwevWS8OhiT3wqqlW7KGPd6m.jpg"
      />
      <h2>
        Fall
      </h2>
      <p>
        <strong>
          2022
        </strong>
      </p>
    </div>
  </div>
</div>
`;
